src/julia_threads.h

```diff
@@ -262,21 +254,17 @@ typedef struct _jl_task_t {
     // === 64 bytes (cache line)
     // timestamp this task finished (i.e. entered state DONE or FAILED).
     _Atomic(uint64_t) finished_at;
+    _Atomic(uint8_t) _state;
+    uint8_t sticky; // record whether this Task can be migrated to a new thread
+    uint16_t priority;
+    _Atomic(uint8_t) _isexception; // set if `result` is an exception to throw or that we exited with
+    // flag indicating whether or not to record timing metrics for this task
+    uint8_t metrics_enabled;
```

Comment:
How about grouping `reentrant_timing` (which needs 4 bits), `metrics_enabled` (one bit) and `sticky` (also one bit) into a single byte. That frees up one byte here, so `threadpoolid` could then be moved here instead of dangling at the end.

Of course that then makes access to these fields a bit more annoying. Not sure how problematic that is?

Reply:
To be clear, I meant a C bitfield, i.e.
```C
    struct {
        uint8_t reentrant_timing : 4;
        uint8_t sticky : 1;
        uint8_t metrics_enabled : 1;
    };
   int8_t threadpoolid;
```